misk-crypto/src/main/kotlin/misk/crypto/pgp/PgpEncoder.kt

```diff
@@ -0,0 +1,27 @@
+package misk.crypto.pgp
+
+import org.bouncycastle.openpgp.PGPPublicKey
+
+interface PgpEncoder {
```

Comment:
@yoavamit brought up a good point on my branch that perhaps this should really be two different interfaces: an encryptor and an encoder.

I guess in this case we'd always create an encoder for a public key, and always create a decryptor for a private key? Does that seem legit?

Reply:
I implemented it as suggested. I think it's better this way.